sql/importer: TestMultiNodeExportStmt failed

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4585814&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4585814&tab=artifacts#/) on release-22.1 @ [2d23da1dd4f7129314efadf2ef053cdf146c467e](https://github.com/cockroachdb/cockroach/commits/2d23da1dd4f7129314efadf2ef053cdf146c467e):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt3197307317
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (160): pq: could not find a better lease transfer target for r45
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (4.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
/cc @cockroachdb/bulk-io
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13860

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4596675&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4596675&tab=artifacts#/) on release-22.1 @ [2d49a7bd0d1ebeff93277233695411d25b2a5d39](https://github.com/cockroachdb/cockroach/commits/2d49a7bd0d1ebeff93277233695411d25b2a5d39):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt918685596
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (160): pq: unable to find store 3 in range r45:/{Table/106/1/100-Max} [(n1,s1):1, (n4,s4):2, (n5,s5):3, next=6, gen=15, sticky=9223372036.854775807,2147483647]
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (5.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4607717&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4607717&tab=artifacts#/) on release-22.1 @ [d7252106b166f23d791a4d677afdbc46b6b059ae](https://github.com/cockroachdb/cockroach/commits/d7252106b166f23d791a4d677afdbc46b6b059ae):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt214944841
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (160): pq: could not find a better lease transfer target for r45
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (6.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4622581&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4622581&tab=artifacts#/) on release-22.1 @ [a25a145510d5d0bc7570db327a21cd3fcde21980](https://github.com/cockroachdb/cockroach/commits/a25a145510d5d0bc7570db327a21cd3fcde21980):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt278446073
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (80): pq: unable to find store 1 in range r46:/Table/106/1/{60-100} [(n5,s5):4, (n2,s2):2, (n4,s4):5, next=6, gen=16, sticky=9223372036.854775807,2147483647]
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (8.41s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4635225&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4635225&tab=artifacts#/) on release-22.1 @ [ec0a78bbf61f7d71fe4222cc1c11d262936df2e3](https://github.com/cockroachdb/cockroach/commits/ec0a78bbf61f7d71fe4222cc1c11d262936df2e3):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt1180491496
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (160): pq: could not find a better lease transfer target for r45
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (6.75s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4647893&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4647893&tab=artifacts#/) on release-22.1 @ [26c05e15d3752f47a91592b5b870360c30c57dd5](https://github.com/cockroachdb/cockroach/commits/26c05e15d3752f47a91592b5b870360c30c57dd5):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt1329262599
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (180): pq: could not find a better lease transfer target for r55
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (4.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/importer.TestMultiNodeExportStmt [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4661659&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4661659&tab=artifacts#/) on release-22.1 @ [face5a80eef3bc0e65d31fc411a1c37a18ca9503](https://github.com/cockroachdb/cockroach/commits/face5a80eef3bc0e65d31fc411a1c37a18ca9503):


```
=== RUN   TestMultiNodeExportStmt
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/3e94a3d804ab82b4843bb87c9154ed0f/logTestMultiNodeExportStmt4068486018
    test_log_scope.go:80: use -show-logs to present logs inline
    exportcsv_test.go:65: executing ALTER TABLE bank SPLIT AT VALUES (160): pq: could not find a better lease transfer target for r45
    panic.go:642: -- test log scope end --
--- FAIL: TestMultiNodeExportStmt (4.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #76922 sql/importer: TestMultiNodeExportStmt failed [C-test-failure O-robot T-bulkio T-kv branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMultiNodeExportStmt.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replica_command.go
+++ b/pkg/kv/kvserver/replica_command.go
@@ -1210,11 +1210,20 @@ func (r *Replica) maybeTransferLeaseDuringLeaveJoint(
 	}
 
 	if voterIncomingTarget == (roachpb.ReplicaDescriptor{}) {
-		// Couldn't find a VOTER_INCOMING target. This should not happen since we only go into the
-		// JOINT config if the leaseholder is being removed, when there is a VOTER_INCOMING replica.
-		// Killing the leaseholder to force lease transfer.
-		log.Fatalf(ctx, "no VOTER_INCOMING to transfer lease to. "+
-			"Range descriptor: %v", desc)
+		// Couldn't find a VOTER_INCOMING target. When the leaseholder is being
+		// removed, we only enter a JOINT config if there is a VOTER_INCOMING
+		// replica. We also do not allow a demoted replica to accept the lease
+		// during a JOINT config. However, it is possible that our replica lost
+		// the lease and the new leaseholder is trying to remove it. In this case,
+		// it is possible that there is no VOTER_INCOMING replica.
+		// We check for this case in replica_raft propose, so here it is safe
+		// to continue trying to leave the JOINT config. If this is the case,
+		// our replica will not be able to leave the JOINT config, but the new
+		// leaseholder will be able to do so.
+		log.Infof(ctx, "no VOTER_INCOMING to transfer lease to. This replica probably lost the lease,"+
+			" but still thinks its the leaseholder. In this case the new leaseholder is expected to "+
+			"complete LEAVE_JOINT. Range descriptor: %v", desc)
+		return nil
 	}
 	log.VEventf(ctx, 5, "current leaseholder %v is being removed through an"+
 		" atomic replication change. Transferring lease to %v", r.String(), voterIncomingTarget)

